### PR TITLE
Np 25482 store cristin id to nva id in s3

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

**Review by ketilaa — APPROVED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 1]
```diff
@@ -54,117 +56,153 @@ public class CristinEntryEventConsumer extends EventHandler<EventReference, Publ
...
     @JacocoGenerated
     private static AmazonDynamoDB defaultDynamoDbClient() {
         return AmazonDynamoDBClientBuilder
                    .standard()
                    .withRegion(ApplicationConstants.AWS_REGION.id())
                    .build();
     }
-    
+
+    private static Optional<String> getCristinIdentifier(Publication publication) {
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:113]:
```
    private static Optional<String> getCristinIdentifier(Publication publication) {
        return publication.getAdditionalIdentifiers()
                   .stream()
                   .filter(CristinEntryEventConsumer::isCristinAdditionalIdentifier)
                   .findFirst()
                   .map(AdditionalIdentifier::getValue);
    }
```

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 2]
```diff
@@ -54,117 +56,153 @@ public class CristinEntryEventConsumer extends EventHandler<EventReference, Publ
...
+                       additionalIdentifier -> isCristinAdditionalIdentifier(additionalIdentifier))
+                   .findFirst()
+                   .map(additionalIdentifier -> additionalIdentifier.getValue());
+    }
+
+    private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additionalIdentifier) {
+        return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
+    }
+
+    private Publication storePublicationIdentifierAndCristinIdInS3(Publication publication,
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:126]: persistCristinIdentifierInFileNamedWithPublicationIdentifier

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 3]
```diff
@@ -54,117 +56,153 @@ public class CristinEntryEventConsumer extends EventHandler<EventReference, Publ
...
+        var cristinIdentifier =
+            getCristinIdentifier(publication);
+        var fileUri = constructResourcehandleFileUri(eventBody, publication);
+        var s3Driver = new S3Driver(s3Client, fileUri.getUri().getHost());
+        attempt(() -> s3Driver.insertFile(fileUri.toS3bucketPath(),
+                                          cristinIdentifier.orElse(StringUtils.EMPTY_STRING))).orElseThrow();
+        return publication;
+    }
+
+    private UriWrapper constructResourcehandleFileUri(FileContentsEvent<JsonNode> eventBody, Publication publication) {
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:137]: Resourcehandle? Not sure what this means

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 4]
```diff
@@ -134,221 +136,226 @@ void shouldReturnAnNvaPublicationEntryWhenInputIsEventWithCristinResult() throws
...
         injectValuesThatAreCreatedWhenSavingInDynamo(actualPublication, expectedPublication);
-        
+
         assertThat(actualPublication, is(equalTo(expectedPublication)));
         assertThat(actualPublication.getStatus(), is(equalTo(PublicationStatus.PUBLISHED)));
     }
-    
+
+    @Test
+    void shouldStoreNvaPuclicationIdAndCristinIdInS3() throws IOException {
```
Comment (brinxmat) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:148]: Publication

Comment (brinxmat) [reply cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:148]: I guess it is "shouldPersistCristinIdInFileNamedWithPublicationIdentifier"

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 5]
```diff
@@ -384,32 +391,45 @@ void shouldThrowExceptionWhenInputContainsUnknownProperty() throws IOException {
...
-    
+
+    private static <T> FileContentsEvent<T> createEventBody(T cristinObject) {
+        return new FileContentsEvent<>(randomString(), EVENT_SUBTOPIC, randomUri(), Instant.now(),
+                                       cristinObject);
+    }
+
+    private static JavaType constructImportResultJavaType() {
+
+        JavaType fileContentsType = eventHandlerObjectMapper.getTypeFactory()
```
Comment (brinxmat) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:421]: Use var

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 1 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:114]: Fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 3 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: Ups. copy-paste from brage-import. Does "constructSuccessFileUri" sound better? It's more inline with the almost identical function "constructErrorFileUri".

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java (hunk 4 above)
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java]: Fikset

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java (hunk 5 above)
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java]: Fikset

**Review by brinxmat — APPROVED** [review]

**Pull request merged**